kolibri/core/assets/src/utils/UserType.js

```diff
@@ -4,7 +4,7 @@ export default function UserType(userObject) {
   if (userObject.is_superuser) {
     return UserKinds.SUPERUSER;
   }
-  if (!userObject.roles.length) {
+  if (!userObject.roles || !userObject.roles.length) {
```

Comment:
No need to change but could also do:
```
if (!userObject?.roles.length) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

(I feel like you used this in the `useBaseSearch` refactor work)